boot-uml.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+BASE=$(dirname "$(readlink -f "$0")")
+source "$BASE"/utils.sh
+
+function parse_parameters() {
+    while (($#)); do
+        case $1 in
+            -i | --interactive | --shell)
+                kernel_args=(init=/bin/sh)
+                ;;
+            -k | --kernel-location)
+                shift
+                KERNEL_LOCATION=$1
+                ;;
+        esac
+        shift
+    done
+}
+
+function reality_check() {
+    [[ -z $KERNEL_LOCATION ]] && die "Kernel image or kernel build folder ('-k') is required but not specified!"
+    KIMAGE=linux get_full_kernel_path
+}
+
+function decomp_rootfs() {
+    rootfs=$BASE/images/x86_64/rootfs.ext4
+    rm -rf "$rootfs"
+    zstd -q -d "$rootfs".zst -o "$rootfs"
+}
+
+function execute_kernel() {
+    # exec is needed to avoid a "Killed" message when the kernel shuts down.
```

Comment:
Please:
1. file an issue about this
2. add a todo linking to the issue above the use of `exec` in this file.

You mentioned that you found a kernel regression about this; let's get an issue on file that can be used to track kernel fixes, and remind us to drop `exec` eventually.

Reply:
> Please:
> 
> 1. file an issue about this

Sure, I will file an issue in this repo rather than the main one, as this regression is not clang specific.

> 2. add a todo linking to the issue above the use of `exec` in this file.
> 
> You mentioned that you found a kernel regression about this; let's get an issue on file that can be used to track kernel fixes, and remind us to drop `exec` eventually.

I will send the report shortly. I would like to see the report addressed before merging this, so we can drop `exec` from the initial commit. We will see how fast the regression is fixed upstream though.

Reply:
We don't know how long it will take to get a fix in mainline; let's land then remove the use of exec later?

Reply:
since https://github.com/ClangBuiltLinux/continuous-integration2/pull/337/commits/03f113e7a9a135ed5151f069d3a0eab72a6f3b82 does add ARCH=um coverage for mainline.

Reply:
Alright, sure. Let me update the comment to include the link to the issue in the repo.